src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -156,18 +157,6 @@ public TaskResult call() {
                         MetricsLevel.SUMMARY);
             }
 
-            if (localReason == ShutdownReason.TERMINATE) {
-                LOG.debug("Looking for child shards of shard " + shardInfo.getShardId());
-                // create leases for the child shards
-                TaskResult result = shardSyncStrategy.onShardConsumerShutDown(latestShards);
```

Comment:
There were lots of placing we were passing around latestShards as an optimization (ShardSyncTaskManager, ShardSyncer, ShardSyncTask). AFAIK this is the only place we ever set it to be non-null, correct me if I'm wrong. We should probably remove this from all those places if we're never setting it to be non-null. That can be outside the scope of this PR though, would make the diff noisier.

Reply:
Yeah, it's a good point that we no longer need this parameter passed into the onShardConsumerShutDown method. I will change the interface.

Reply:
To avoid increasing the scope of the current PR, I think it would be better to make this refactor change in a separate PR.